src/call_frame.rs

```diff
@@ -22,6 +23,10 @@ pub struct CallFrame {
     pub gas_left: Saturating<u32>,
     /// Transient storage should be used for temporary storage within a transaction and then discarded.
     pub transient_storage: InMemory,
+    pub address: Address,
+    pub caller: Address,
+    pub code_address: Address,
+    pub context_u128: u128,
```

Comment:
I think this would be better stored on the Context struct, since they do not change between near calls

Reply:
You are right, I've moved them there.